### ISSUE TITLE
[REFACTOR] Trace Analytics - lift data-source-specific conditions into a global filter

### ISSUE THREAD

**Issue by pjfitzgibbons** [opensearch-project/dashboards-observability#797]
**What is the issue?**
Current Trace Analytics display logic requires testing data-source "mode" on every render.
This code is currently embedded within the components.
https://github.com/opensearch-project/dashboards-observability/pull/551#discussion_r1245482283

**What is the expected code?**
Lift this conditional pattern into a global object/function/class that can be queried for truth condition.
This will also help the code "tell the story" through effective naming.

**What is your host/environment?**
 - Plugins

**Do you have any screenshots?**

**Do you have any additional context?**
Add any other context about the problem.

**Comment by pjfitzgibbons** [comment]
@derek-ho Could you describe in a bit more detail why the condition linked-to in description exists?  What is the truth of the condition accomplishing?
It's reportedly used in many places.  Is there a common pattern to the conditional test? - common arguments?

**Comment by derek-ho** [comment]
Yes I was planning on doing this - there were several options being thrown about including a section in advanced settings specifically for observability - related things this being one of them. Not sure where that landed but I can check back in
